plugins/transport/socket/main.go

```diff
@@ -201,10 +199,13 @@ func (s *Socket) ReceiveData(maxBuffSize int64, done chan bool, pc net.Conn, w t
 		}
 
 		if s.conf.Type == tcp {
-			remainingMsg, err = s.WriteTCPMsg(w, msgBuffer, n)
+			parsed, err := s.WriteTCPMsg(w, msgBuffer, n)
 			if err != nil {
+				s.logger.Errorf(err, "error, while parsing messages")
 				return
 			}
+			remainingMsg = make([]byte, int64(n) - parsed)
```

Comment:
Okay, this is making more sense. I was reading over this code just a few minutes ago trying to formulate a comment to explain exactly what was bothering me about it. Seems you figured it out by running it more; good job! :)